generic.c

```diff
@@ -164,3 +164,30 @@ int createFile(const char * file, const char * buff, size_t size)
 	return SUCCESS;
 }
 
+/*
+ *returns a new pointer to an array with new length
+ *@param arr , a pointer to the array, will be reallocated to have desired size or NULL if error
+ *@param new_length , the desired length of the array
+ *@return 0 for success or ALLOC_FAIL if fail (memory will be freed in this case)
+ */
+int reallocArray(void **arr, size_t new_length)
+{
+	void *old_arr;
+	size_t new_size;
+	//if realloc returns null it does not free memory so we must keep a pointer to it
+	old_arr = *arr;
+	//check if requested size is too big
+	if (__builtin_mul_overflow(new_length, sizeof(*arr), &new_size)) {
```

Comment:
This will always be `void *`, so this function appears to assume the array we are reallocating has an element size of `sizeof(void *) -> 8 (probably)`